### PR TITLE
use make update-go-verison to update go version in Dockerfile

### PR DESCRIPTION
**Pull request by duanmengkk**
#### What type of PR is this?

release/none 

**What this PR does / why we need it**:
release/none

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ty-dc** [comment]
It seems that there is this tag environment variable, but it got a null value in CI.

```
echo "tag=$(git ls-tree --full-tree HEAD -- ./images/spiderpool-base | awk '{ print $3 }')" >> $GITHUB_ENV
```

**Comment by duanmengkk** [comment]
> It seems that there is this tag environment variable, but it got a null value in CI.
> 
> ```
> echo "tag=$(git ls-tree --full-tree HEAD -- ./images/spiderpool-base | awk '{ print $3 }')" >> $GITHUB_ENV
> ```

yes, I'm not very familiar with this area, could you tell me how to fix the CI errors? @ty-dc

**Comment by ty-dc** [comment]
> > It seems that there is this tag environment variable, but it got a null value in CI.
> > ```
> > echo "tag=$(git ls-tree --full-tree HEAD -- ./images/spiderpool-base | awk '{ print $3 }')" >> $GITHUB_ENV
> > ```
> 
> yes, I'm not very familiar with this area, could you tell me how to fix the CI errors? @ty-dc

Maybe setting the fetch-depth field to 0 will make CI pass, you can try it.
```
      - name: Checkout Source Code
        uses: actions/checkout@v4
        with:
          persist-credentials: false
          fetch-depth: 0
          ref: ${{ env.event_ref }}
```

File: .github/workflows/build-image-base.yaml
Comment (ty-dc) [inline .github/workflows/build-image-base.yaml:1]: This `action` looks like it hasn't been run in a while.

**Comment by ty-dc** [comment]
Confusingly, the CI didn't trigger. We will focus on CI status.

File: .github/workflows/build-image-base.yaml [hunk 1]
```diff
@@ -88,7 +88,7 @@ jobs:
         id: tag-in-repositories
         shell: bash
         run: |
-          if docker buildx imagetools inspect ${{ env.ONLINE_REGISTER }}/${{ env.IMAGE_REPO }}/${{ env.IMAGE_NAME }}:${{  steps.base_tag.outputs.tag }} &>/dev/null; then
+          if docker buildx imagetools inspect ${{ env.ONLINE_REGISTER }}/${{ env.IMAGE_REPO }}/${{ env.IMAGE_NAME }}:${{ steps.base_tag.outputs.tag }} &>/dev/null; then
```
Comment (ty-dc) [inline .github/workflows/build-image-base.yaml:91]:
```
{{ steps.base_tag.outputs.tag }} --> {{ env.tag }}
```

Comment (duanmengkk) [reply .github/workflows/build-image-base.yaml]: It seems to not work

**Comment by duanmengkk** [comment]
My code is
image
and the log in ci is 
image

It seems my code is not work

**Comment by ty-dc** [comment]
I tried it and it seems to work.

```
/usr/bin/docker buildx build --file ./images/spiderpool-base/Dockerfile --iidfile /home/runner/work/_temp/docker-actions-toolkit-jDPRjT/iidfile --platform linux/amd64,linux/arm64 --provenance false --tag ghcr.io/ty-dc/spiderpool/spiderpool-base:44bfacb93132697eacbdd5faddcfc8b38894ead3 --metadata-file /home/runner/work/_temp/docker-actions-toolkit-jDPRjT/metadata-file --push ./images/spiderpool-base
```

See if it helps you? refer to [here](https://github.com/ty-dc/spiderpool/commit/ee3f9e54df66bdea1fcdf5ad4454ce77b4568d63)

**Comment by ty-dc** [comment]
In PR #3316 I tested this phenomenon. This is a CI bug, but due to the default nature of the action: "The submitted PR contains workflow changes, but when running CI, the process triggered by the workflow of the master branch is still used."

I submitted a PR to pre-fix this issue, you can rebase the main branch later.  thanks .

**Comment by ty-dc** [comment]
Hi, a friendly push @duanmengkk , the problem is fixed, you can rebase main, thank you.

**Review by ty-dc — APPROVED** [review]

**Pull request closed without merging**